app_allocator/classes/feature_distribution.py

```diff
@@ -0,0 +1,57 @@
+from csv import DictReader
+from random import random
+from app_allocator.classes.feature import Feature
+
+
+JUDGE_TYPE = "judge"
+MATCHING_TYPE = "matching"
+NAME_TYPE = "name"
+READS_TYPE = "reads"
+
+
+class FeatureDistribution(object):
+    all_distributions = {}
+    matching_distributions = []
+
+    def __init__(self, type, name, values=None):
+        self.feature = Feature.find_feature(type, name)
+        self.values = values
+        self.weights = {}
+        self.total = 0
+        FeatureDistribution.all_distributions[name] = self
+        if type == MATCHING_TYPE:
+            FeatureDistribution.matching_distributions.append(self)
+
+    @classmethod
+    def load_distributions(cls, filename):
+        file = open(filename)
+        cls.read_from_file(file)
+        file.close()
+
+    @classmethod
+    def read_from_file(cls, file):
+        reader = DictReader(file)
+        for row in reader:
+            name = row["name"]
+            distribution = FeatureDistribution.all_distributions.get(name)
```

Comment:
Should this read `distribution = cls.all_distributions.get(name)`?

Reply:
Sure.  Done.